cmr-exchange/metadata-proxy/src/cmr/metadata/proxy/concepts/variable.clj

```diff
@@ -36,24 +39,24 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 (defn build-query
-  [variable-ids]
+  "Build variable search query based on variable-info, which could be
+  list of concept-ids, or aliases. search-constraint could be either
+  concept_id[] or alias[]." 
+  [variable-info search-constraint]
   (string/join
-   "&"
-   (conj
-    (map #(str (codec/url-encode "concept_id[]")
-               "=" %)
-         variable-ids)
-    (str "page_size=" (count variable-ids)))))
+    "&"
+    (conj
+      (map #(str (codec/url-encode search-constraint) "=" %) variable-info)
+      (str "page_size=" (count variable-info)))))
 
 (defn async-get-metadata
-  "Given a 'params' data structure with a ':variables' key (which may or may
-  not have values) and a list of all collection variable-ids, return the
-  metadata for the passed variables, if defined, and for all associated
-  variables, if params does not contain any."
```

Comment:
The old code only searched for the variables that matched the variable concept-ids before.  it never returned "all associated variables, if params doesn't not contain any."  Is it supposed to return all? if so, the old code was wrong. I'm assuming the old code was right and just the doc was wrong.